wandb/integration/catboost/catboost.py

```diff
@@ -0,0 +1,176 @@
+"""
+catboost init
+"""
+
+from pathlib import Path
+from types import SimpleNamespace
+from typing import Union
+
+from catboost import CatBoostClassifier, CatBoostRegressor  # type: ignore
+import wandb
+
+
+class WandbCallback:
+    """`WandbCallback` automatically integrates CatBoost with wandb.
+
+    Arguments:
+        - metric_period: (int) if you are passing `metric_period` to your CatBoost model please pass the same value here (default=1).
+
+    Passing `WandbCallback` to CatBoost will:
+    - log training and validation metrics at every `metric_period`
+    - log iteration at every `metric_period`
+
+    Example:
+        ```
+        train_pool = Pool(train[features], label=train["label"], cat_features=cat_features)
+        test_pool = Pool(test[features], label=test["label"], cat_features=cat_features)
+
+        model = CatBoostRegressor(
+            iterations=100,
+            loss_function="Cox",
+            eval_metric="Cox",
+        )
+
+        model.fit(
+            train_pool,
+            eval_set=test_pool,
+            callbacks=[WandbCallback()],
+        )
+        ```
+    """
+
+    def __init__(self, metric_period: int = 1):
+        if wandb.run is None:
+            raise wandb.Error("You must call `wandb.init()` before `WandbCallback()`")
+
+        self.metric_period: int = metric_period
+
+    def after_iteration(self, info: SimpleNamespace) -> bool:
+        if info.iteration % self.metric_period == 0:
+            for data, metric in info.metrics.items():
+                for metric_name, log in metric.items():
+                    wandb.log({f"{data}-{metric_name}": log[-1]}, commit=False)
```

Comment:
```suggestion
                    wandb.run._log({f"{data}-{metric_name}": log[-1]}, commit=False)
```
so maybe we can create an internal function in the run class:
I'm adding in my PR:
https://github.com/wandb/client/pull/3137/files#diff-b63db14fd18e0e4d240ec68e625f5420067376cb574e03f383ca7a29e7bfce4bR1134

Reply:
Should I commit this now?

Reply:
add a todo... sorry that I used the suggest...
it won't work (we don't have internal function)